File: week-2/set-mismatch/set-mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    const hashtable = {};
    const res = [];
    for (let i of nums) {
        if (i in hashtable)
            res.push(i)
        hashtable[i] = true;
    }
    
    for (let i=1; i<=nums.length; i++) {
        if (!(i in hashtable)) {
            res.push(i);
            break;
        }
    }
    
    return res;
};

File: week-2/set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        hashtable = {}
        res = []
        for i in nums:
            if i in hashtable:
                res.append(i)
            hashtable[i] = True
        
        for i in range(1, len(nums)+1):
            if i not in hashtable:
                res.append(i)
                break
                
        return res
